### PR TITLE
refactor BTCDepositAddressDeriver

### PR DESCRIPTION
**Pull request by szhygulin**
refactor BTCDepositAddressDeriver

### CODE REVIEW

File: src/BTCDepositAddressDeriver.sol [hunk 1]
```diff
@@ -31,48 +44,71 @@ contract BTCDepositAddressDeriver {
...
     }
 
+    // Set Validators' joint pubkeys and network prefix, must be called after contract deployment
+    function setSeed(
+        string calldata _btcAddr1,
+        string calldata _btcAddr2,
+        BitcoinNetwork _network
+    ) public virtual {
+
+        string memory _hrp;
```
Comment (bullet-tooth) [inline src/BTCDepositAddressDeriver.sol:54]: nit: if remove this var and assign directly to `networkHrp`  we will reduce gas footprint?

File: script/SetSeed.s.sol [hunk 2]
```diff
@@ -9,23 +9,28 @@ import {BTCDepositAddressDeriver} from "../src/BTCDepositAddressDeriver.sol";
...
         console.log("BTC_ADDR2:", btcAddr2);
 
         BTCDepositAddressDeriver deriver = BTCDepositAddressDeriver(
             contractAddress
         );
+
+        // set validators' pubkeys and network prefix
         vm.startBroadcast();
-        deriver.setSeed(btcAddr1, btcAddr2, "bcrt");
+        deriver.setSeed(btcAddr1, btcAddr2, BTCDepositAddressDeriver.BitcoinNetwork.REGTEST);
```
Comment (bullet-tooth) [inline script/SetSeed.s.sol:33]: Pick up network from env as well?

File: src/BTCDepositAddressDeriver.sol [hunk 3]
```diff
@@ -31,48 +44,71 @@ contract BTCDepositAddressDeriver {
...
+
+        if (_network == BitcoinNetwork.TESTNET) {
+            _hrp = 'tb';
+        } else if (_network == BitcoinNetwork.MAINNET) {
+            _hrp = 'bc';
+        } else if (_network == BitcoinNetwork.REGTEST) {
+            _hrp = 'brct';
+        }
```
Comment (bullet-tooth) [inline src/BTCDepositAddressDeriver.sol:62]: looks like it's a good candidate to extract to a separate public method

File: script/SetSeed.s.sol (hunk 2 above)
Comment (szhygulin) [reply script/SetSeed.s.sol:33]: fixed. But needed to read network as int and convert, since picking up enums is not supported

**Review by bullet-tooth — APPROVED** [review]

**Pull request merged**